actors/evm/tests/storage_footprint.rs

```diff
@@ -0,0 +1,242 @@
+use std::fs::File;
+use std::io::Write;
+use std::sync::Arc;
+
+use ethers::contract::Lazy;
+use ethers::prelude::abigen;
+use ethers::providers::{MockProvider, Provider};
+use fvm_ipld_blockstore::tracking::BSStats as BlockstoreStats;
+use fvm_shared::address::Address;
+
+mod env;
+
+use env::{TestContractCall, TestEnv};
+use serde_json::json;
+
+// Generate a statically typed interface for the contract.
+abigen!(StorageFootprint, "./tests/contracts/StorageFootprint.abi");
+
+// Alternatively we can generate the ABI code as follows:
+// ```
+//     ethers::prelude::Abigen::new("StorageFootprint", "./tests/contracts/StorageFootprint.abi")
+//         .unwrap()
+//         .generate()
+//         .unwrap()
+//         .write_to_file("./tests/storage_footprint_abi.rs")
+//         .unwrap();
+// ```
+
+// The owner doesn't matter in these tests, so just using the same value that the other tests use, for everything.
+const OWNER: Address = Address::new_id(100);
+
+static CONTRACT: Lazy<StorageFootprint<Provider<MockProvider>>> = Lazy::new(|| {
+    // The owner of the contract is expected to be the 160 bit hash used on Ethereum.
+    // We're not going to use it during the tests.
+    let owner_hex = format!("{:0>40}", hex::encode(OWNER.payload_bytes()));
+    let address = owner_hex.parse::<ethers::core::types::Address>().unwrap();
+    // A dummy client that we don't intend to use to call the contract or send transactions.
+    let (client, _mock) = Provider::mocked();
+    StorageFootprint::new(address, Arc::new(client))
+});
+
+/// Create a fresh test environment.
+fn new_footprint_env() -> TestEnv {
+    let mut env = TestEnv::new(OWNER);
+    env.deploy(include_str!("contracts/StorageFootprint.hex"));
+    env
+}
+
+struct Measurements {
+    scenario: String,
+    values: Vec<serde_json::Value>,
+}
+
+impl Measurements {
+    pub fn new(scenario: String) -> Self {
+        Self { scenario, values: Vec::new() }
+    }
+
+    pub fn record(&mut self, series: u8, i: u32, stats: BlockstoreStats) {
+        // Not merging `i` into `stats` in JSON so the iteration appear in the left.
+        let value = json!({
+            "i": i,
+            "series": series,
+            "stats": {
+                "get_count": stats.r,
+                "get_bytes": stats.br,
+                "put_count": stats.w,
+                "put_bytes": stats.bw,
+            }
+        });
+        self.values.push(value);
+    }
+
+    pub fn export(self) -> Result<(), std::io::Error> {
+        let dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
```

Comment:
nit: this should be `env!` as there isn't a guarantee that this will be set at _runtime_, IIRC.